### ISSUE TITLE
undefined does not have a method "startsWith"

### ISSUE THREAD

**Issue by algesten** [raine/replem#4]
When I try using a package I never used before, I get this:

```
$ replem tlbx:T
[TypeError: undefined does not have a method named "startsWith"]
```

I `console.log` [_resolved](https://github.com/raine/replem/blob/master/src/index.js#L64) to find this:

```
sultana$ replem tlbx:T
{ _from: 'tlbx',
    _resolved: undefined,
    name: 'tlbx',
    version: '0.0.6' }
```

And I trace this back to `.replem/node_modules/tlbx/package.json`

```
$ grep resolved ~/.replem/node_modules/tlbx/package.json 
$ grep resolved ~/.replem/node_modules/fnuc/package.json 
  "_resolved": "https://registry.npmjs.org/fnuc/-/fnuc-1.1.0.tgz",
```

So it seems for all npm packages, we don't get the `_resolved` bit?

**Comment by raine** [comment]
Hmm, I get it.

```
~ ❯❯❯ grep resolved ~/.replem/node_modules/tlbx/package.json
  "_resolved": "https://registry.npmjs.org/tlbx/-/tlbx-0.0.6.tgz"
```

Can you reproduce it?

**Comment by algesten** [comment]
it happened once for me before with fnuc. and now with this one. I guess I `rm -rf` stuff in between. I'm on an almost okay internet connection. The odd packet loss, though surely this stuff have tcp resends an all sorts of error correction... I try some more.

**Comment by raine** [comment]
``` diff
diff --git a/src/index.js b/src/index.js
index 92d139c..221f693 100644
--- a/src/index.js
+++ b/src/index.js
@@ -54,7 +54,7 @@ const installMultiple = (packages, cb) => {

 const getResolvedSha = pipe( split('#'), last, take(7) );
 const formatVersion = (resolved, version) =>
-  startsWith('git://', resolved)
+  resolved && startsWith('git://', resolved)
     ? '#' + getResolvedSha(resolved)
     : '@' + version;
```

This would be a possible fix but I'd like to know how it can end up in a state like this.

Perhaps npm v3 would not have this issue.

**Comment by raine** [comment]
Fixed in v1.8.3.

**Comment by davidchase** [comment]
@raine i tried using your replem and i get a similar error:

``` sh
"https://registry.npmjs.org/ramda/-/ramda-0.18.0.tgz" does not have a method named "startsWith"
```

my npm is `3.3.5` and a buddy of mine had the same issue with npm at `2.13.4` with replem at `1.9.0`
on a MacOSX

any thoughts ?

**Comment by raine** [comment]
Thanks @davidchase, fixed in 425dc04811fcfe79e056d5a3a8a49fbbbc6b2b4c and 1.9.1

I hadn't noticed this because I run replem with node v4 usually, and I thought babel would add ES6 things for native types.

**Comment by davidchase** [comment]
Thanks for the quick update... We are still behind on node v4 at my company was going to try to use your repl on my own Mac which has node v4
